product-pricing-service/src/main/java/com/hedvig/contract/graphql/resolvers/ContractQuery.kt

```diff
@@ -40,7 +40,18 @@ class ContractQuery(
   fun getInsuranceCost(env: DataFetchingEnvironment): InsuranceCost? {
     val memberId = env.getToken()
     val totalCost = premiumService.getTotalPremiumOfContractsWithActiveActiveInFutureOrPendingAgreementsByHolderMemberId(memberId)
-    val bundleDiscount = bundleDiscountService.getBundleDiscountOrNullForPeriod(memberId, YearMonth.now(), true)
+    //fixme: This is a lot of code in the wrong place but lets fix that by breaking out graphQL from contracts and campaign to it own module
+    val firstMasterInception = contractService.getContractStatesByHolderMemberId(memberId)
+      .filter { !it.isTerminated }
```

Comment:
Unfortunately this is not sufficent (agreements is a bastard really). Having a termination date is not enough to know if it is terminated today. I think this is more correct however.
```suggestion
      .filter { !it.getContractStatus != ContractStatus.TERMINATED}
```

We should probably rename "isTerminated" to "hasTerminationDate" instead

Reply:
Okay good to know!